services/fib2/src/main.rs

```diff
@@ -385,13 +387,19 @@ extern "C" fn accumulate() -> u64 {
 
     // write yield
     if n % 3 == 0 {
+        if n != 9 { // n=3,6 should go through even though there is a panic, 9 does not.
+            let gas_result = unsafe { checkpoint() };
```

Comment:
Note that in the 9 case NO WRITES HAPPEN but on the 3, 6 case WRITES HAPPEN.

This is because checkpoint copies X into Y.